lisp/init-selectrum.el

```diff
@@ -0,0 +1,52 @@
+;;; init-selectrum.el --- Config for selectrum       -*- lexical-binding: t; -*-
+;;; Commentary:
+;;; Code:
+
+(when (maybe-require-package 'orderless)
+  (setq completion-styles '(orderless)))
+
+(when (require-package 'selectrum)
+  (add-hook 'after-init-hook 'selectrum-mode)
+
+  (when (maybe-require-package 'selectrum-prescient)
+    (require 'prescient)
+    (prescient-persist-mode 1)
+    (selectrum-prescient-mode 1)
+    (global-set-key [remap execute-extended-command] nil)
+    (global-set-key [remap execute-extended-command] 'execute-extended-command))
+
+  ;;(defun sanityinc/read-buffer-function (prompt &optional def require-match predicate))
+
+  (when (maybe-require-package 'embark)
+    (define-key selectrum-minibuffer-map (kbd "C-c C-o") 'embark-export)
+    (define-key selectrum-minibuffer-map (kbd "C-c C-c") 'embark-act))
```

Comment:
@oantolin and I seem to prefer `'embark-act-noexit` these days as default, since you can always exit using C-g afterwards.

Furthermore you may want to offer `embark-collect-snapshot`, which gives you access to Embark actions in the buffer, so you stay within Embark in some sense.  In contrast `embark-export` is really an export to something else, dired, occur-mode, ibuffer etc.

Reply:
I'm a fan of `embark-collect-snapshot`, obviously, but I think `embark-export` is a great default:

- It actually just does `embark-collect-snapshot` in most cases, so really the difference is not large.
- In the file and buffer case, where you get a dired or ibuffer buffer, you really don't miss out on any actions, though of course, you have to use those major mode's bindings for them.
- In the cases that export to grep or occur, you wouldn't have had any embark actions anyway other than saving on the kill ring and going to the location (which RET does anyway in grep or occur).

Reply:
Thanks for the embark-act-noexit tip, I'll use that too!

Regarding embark-export, my goal here was to backfill my previous ivy-occur workflow, which was very grep-centric. I tried embark-collect-snapshot but it wasn't as direct for this purpose.

I think it'll take a while to build habits of using much of embark, to be honest. ivy-occur was the only non-default ivy action I ever used. :-)

Reply:
@oantolin I just noticed this is the new `embark-occur`, I noticed it doesn't exit the minibuffer when you call it. For embark-export the minibuffer is still exited it seems, but when there is no export and the the default embark-collect-snapshot gets invoked it is kept open, is that on purpose?

Reply:
Yes, it is on purpose @clemera, but I'm not so sure it is a good idea. The idea is that you call it from either `embark-act` or from `embark-act-noexit` (it's bound to `S` in the general action map), according to whether you want it to exit or not.

Reply:
I see, thanks, maybe make exit the default so it is the same for export/collect and then if one doesn't want to exit it could still be called from `embark-act-noexit`?